internal/settings/manager.go

```diff
@@ -0,0 +1,38 @@
+package settings
```

Comment:
Do we even need a "Manager" abstraction? The config is just a k8s object, what if we just lean into that and do k8s client "gets" where appropriate to pull the current config?

cc: @zachaller

Reply:
Originally I was just doing a Get but @agaudreault brought a good point (https://github.com/argoproj-labs/gitops-promoter/pull/224#discussion_r2007998761) that we will probably doing some processing on the config such as merging it with a more specific config. But I do agree that it's an unnecessary abstraction until then.

Reply:
Yeah I feel like I'd rather remove the abstraction until we have the use case to design it around. @agaudreault is that alright?

Reply:
I think the struct allows to centralize the code on how to get the settings. This is very likely to be refactored in the future, so having the code centralized will make it easier to refactor, and allows to mock/abstract the config in the unit tests. 

In doesn't matter if the config is one k8s object, multiple ones, or anything more complex using informers behind the scene. In the end, the promotion config is a `PromotionConfiguration` object and this class abstracts it.

Reply:
> allows to mock/abstract the config logic in the unit tests

Mock APIs are available out of the box for client-go, so I think this would actually make mocks more difficult.

I'm worried that future-us will look at the already half-built abstraction and simply choose to operate within it rather than being open to other ways of handling config. In other words, I'd rather avoid building the design until we have the use case to design around.